src/main/java/org/gridsuite/study/server/service/SupervisionService.java

```diff
@@ -0,0 +1,140 @@
+package org.gridsuite.study.server.service;
+
+import org.gridsuite.study.server.networkmodificationtree.entities.NetworkModificationNodeInfoEntity;
+import org.gridsuite.study.server.repository.networkmodificationtree.NetworkModificationNodeInfoRepository;
+import org.gridsuite.study.server.service.dynamicsimulation.DynamicSimulationService;
+import org.gridsuite.study.server.service.shortcircuit.ShortCircuitService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.client.HttpStatusCodeException;
+
+import java.util.List;
+import java.util.UUID;
+
+import static org.gridsuite.study.server.StudyException.Type.SUPERVSION_ERROR_DELETE_RESULTS;
+import static org.gridsuite.study.server.utils.StudyUtils.handleHttpError;
+
+@Service
+public class SupervisionService {
+    @Autowired
+    private ReportService reportService;
+
+    @Autowired
+    private LoadFlowService loadFlowService;
+
+    @Autowired
+    private DynamicSimulationService dynamicSimulationService;
+
+    @Autowired
+    private SecurityAnalysisService securityAnalysisService;
+
+    @Autowired
+    private SensitivityAnalysisService sensitivityAnalysisService;
+
+    @Autowired
+    private ShortCircuitService shortCircuitService;
+
+    @Autowired
+    private VoltageInitService voltageInitService;
+
+    private final NetworkModificationNodeInfoRepository networkModificationNodeInfoRepository;
+
+    public SupervisionService(NetworkModificationNodeInfoRepository networkModificationNodeInfoRepository) {
+        this.networkModificationNodeInfoRepository = networkModificationNodeInfoRepository;
+    }
+
+    @Transactional
+    public void deleteLoadflowResults() {
+        List<NetworkModificationNodeInfoEntity> nodes = networkModificationNodeInfoRepository.findAllByLoadFlowResultUuidNotNull();
+        nodes.stream().forEach(node -> {
+            node.setLoadFlowResultUuid(null);
+            networkModificationNodeInfoRepository.save(node);
```

Comment:
It 's a transaction you don't need to call save. It will call saveAll maybe itself and it's better !

Reply:
Done